src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/RSACryptoServiceProvider.Unix.cs

```diff
@@ -0,0 +1,241 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+using Internal.NativeCrypto;
+using System.IO;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class RSACryptoServiceProvider : RSA, ICspAsymmetricAlgorithm
+    {
+        private readonly RSA _impl;
+        private bool _publicOnly;
+
+        public RSACryptoServiceProvider()
+            : this(0, true) { }
+
+        public RSACryptoServiceProvider(int dwKeySize)
+            : this(dwKeySize, false) { }
+
+        public RSACryptoServiceProvider(int dwKeySize, CspParameters parameters)
+            : this(dwKeySize, false) { }
+
+        public RSACryptoServiceProvider(CspParameters parameters)
+            : this(0, true) { }
+
+        private RSACryptoServiceProvider(int keySize, bool useDefaultKeySize)
+        {
+            if (keySize < 0)
+                throw new ArgumentOutOfRangeException(nameof(keySize), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            // This class wraps RSA
+            _impl = RSA.Create();
+            _impl.KeySize = useDefaultKeySize ? 1024 : keySize;
+        }
+
+        public CspKeyContainerInfo CspKeyContainerInfo
+        {
+            get { throw new PlatformNotSupportedException(); }
+        }
+
+        public byte[] Decrypt(byte[] rgb, bool fOAEP)
+        {
+            if (rgb == null)
+                throw new ArgumentNullException(nameof(rgb));
+
+            // size check -- must be at most the modulus size
+            if (rgb.Length > (KeySize / 8))
+                throw new CryptographicException(SR.Format(SR.Cryptography_Padding_DecDataTooBig, Convert.ToString(KeySize / 8)));
+
+            return _impl.Decrypt(rgb, fOAEP ? RSAEncryptionPadding.OaepSHA1 : RSAEncryptionPadding.Pkcs1);
+        }
+
+        public override byte[] Decrypt(byte[] data, RSAEncryptionPadding padding)
+        {
+            if (data == null)
+                throw new ArgumentNullException(nameof(data));
+            if (padding == null)
+                throw new ArgumentNullException(nameof(padding));
+
+            if (padding == RSAEncryptionPadding.Pkcs1)
```

Comment:
The was taken from the windows implementation. We can directly flow the padding to _impl.Decrypt\Encrypt but then the semantics will be different than windows. Thoughts?